Kitodo-API/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java

```diff
@@ -44,28 +69,250 @@
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Classpath could not be accessed", e.getMessage());
+        }
+    }
+
+    /**
+     * If the found jar files have frontend files, they will be extracted and copied into the frontend folder
+     * of the core module. Before copying, existing frontend files of the same module will be deleted from the
+     * core module. Afterwards the created temporary folder will be deleted as well.
+     */
+    private void loadFrontendFilesIntoCore() {
+
+        Path moduleFolder = FileSystems.getDefault().getPath(modulePath);
+
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(moduleFolder, "*.jar")) {
+
+            for (Path f : stream) {
+                File loc = new File(f.toString());
+                try (JarFile jarFile = new JarFile(loc)) {
+
+                    if (hasFrontendFiles(jarFile)) {
+
+                        Path temporaryFolder = Files.createTempDirectory(SYSTEM_TEMP_FOLDER, TEMP_DIR_PREFIX);
+
+                        File tempDir = new File(Paths.get(temporaryFolder.toUri()).toAbsolutePath().toString());
+
+                        extractFrontEndFiles(loc.getAbsolutePath(), tempDir);
+
+                        String moduleName = extractModuleName(tempDir);
+                        if (!moduleName.isEmpty()) {
+                            FacesContext facesContext = FacesContext.getCurrentInstance();
+                            HttpSession session = (HttpSession) facesContext.getExternalContext().getSession(false);
+
+                            String filePath = session.getServletContext().getRealPath(File.separator + PAGES_FOLDER)
+                                    + File.separator + moduleName;
+                            FileUtils.deleteDirectory(new File(filePath));
+
+                            String resourceFolder = String.join(File.separator,
+                                    Arrays.asList(tempDir.getAbsolutePath(), META_INF_FOLDER, RESOURCES_FOLDER));
+                            copyFrontEndFiles(resourceFolder, filePath);
+                        } else {
+                            logger.info("No module found in JarFile '" + jarFile.getName() + "'.");
+                        }
+                        FileUtils.deleteDirectory(tempDir);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Classpath could not be accessed", e.getMessage());
+        }
+    }
+
+    /**
+     * Extracts the module name of the current module by finding the pom.properties
+     * in the given temporary folder
+     *
+     * @param temporaryFolder
+     *            folder in which the pom.properties file will be searched for
+     *
+     * @return boolean
+     */
+    private String extractModuleName(File temporaryFolder) throws IOException {
```

Comment:
Return value of method is String but JavaDoc claims boolean. I think that JavaDoc needs an update.